lib/topic_query.rb

```diff
@@ -892,13 +892,22 @@ def remove_muted_tags(list, user, opts = nil)
           list # if viewing the topic list for a muted tag, show all the topics
         else
           muted_tag_ids = TagUser.lookup(user, :muted).pluck(:tag_id)
-          list = list.where("
-            EXISTS (
-              SELECT 1
-                FROM topic_tags tt
-               WHERE tt.tag_id NOT IN (:tag_ids)
-                 AND tt.topic_id = topics.id
-            ) OR NOT EXISTS (SELECT 1 FROM topic_tags tt WHERE tt.topic_id = topics.id)", tag_ids: muted_tag_ids)
+          if SiteSetting.mute_other_present_tags
+            list = list.where("
+              NOT EXISTS(
```

Comment:
A bit concerned about `not exists` performance at scale, have you tested the query against a site like meta? you can do it in data explorer.

Reply:
The query times are very similar.